docs: How to setup monitoring stack? (#480)
kinvolk/lokomotive: docs/how-to-guides/access-monitoring.md

> # Access Monitoring on Lokomotive

Reviewer: As said before, I feel like "how to open Grafana in my browser" is too narrow a topic for a self-contained tutorial :-) I feel like this material should appear at the end of a more complete tutorial which explains how to deploy the Lokomotive monitoring stack. To emphasize, the config reference isn't a tutorial as it doesn't explain **how to do things**.

Since we currently don't seem to have such a tutorial, I suggest we create a "skeleton" for such a tutorial with `TODO` in all the sections you haven't written, include your new sections there and open an issue for completing the guide. You can use [this](https://github.com/kinvolk/lokomotive/blob/master/docs/how-to-guides/authentication-with-dex-gangway.md) how-to guide as an example if it helps you.

Author: I have  added information about how to deploy, PTAL.

## Contents

* [Introduction](#introduction)
* [Prerequisites](#prerequisites)
* [Prometheus](#prometheus)
* [Alertmanager](#alertmanager)
* [Grafana](#grafana)
  * [Using Port Forward](#using-port-forward)
  * [Using Ingress](#using-ingress)

## Introduction

> This guide explains how to access the monitoring stack. Lokomotive ships monitoring stack in the form of prometheus-operator component. Learn more about the prometheus-operator component and its parameters [here](../configuration-reference/components/prometheus-operator.md).

Reviewer: >This guide explains how to access the monitoring stack.

Which monitoring stack? The user has just started reading the document and don't yet know what we are talking about...

>Lokomotive ships monitoring stack in the form of prometheus-operator component. Learn more about the prometheus-operator component and its parameters [here](../configuration-reference/components/prometheus-operator.md).

There are missing articles before the words "monitoring" and "prometheus-operator".

## Prerequisites

> - Deploy the prometheus-operator component in `monitoring` namespace. If different namespace used in the configuration then change the subsequent commands accordingly, `kubectl -n <namespace> ...`.

Reviewer: >Deploy the prometheus-operator component in `monitoring` namespace.

There is a missing article here.

>If different namespace used in the configuration then change the subsequent commands accordingly, `kubectl -n <namespace> ...`.

- Missing article before "different".
- Missing verb before "used" (should be "is used").
- There should be a comma after "configuration" (since the conditional is an introductory phrase) according to our style guide.
- There should not be a "then" after a conditional statement according to our style guide.

I would write the sentence as follows:

>This guide assumes the prometheus-operator component is deployed in a namespace called `monitoring`. If a different namespace is used, replace `monitoring` with that namespace throughout this guide.

> ## Prometheus

Reviewer: ```suggestion
## Accessing Prometheus
```

IMO "Prometheus" isn't descriptive enough as a section title. Ditto similar cases.

> Execute the command to port forward prometheus locally on port `9090`:

Reviewer: ```suggestion
Execute the following command to forward port `9090` locally to the Prometheus pod:
```

```

> kubectl -n monitoring port-forward svc/prometheus-operator-prometheus 9090:9090

Reviewer: Nit: AFAIK you could specify `9090` instead of `9090:9090` with the same effect.

```

> Now open the following URL: [http://localhost:9090](http://localhost:9090).

Reviewer: I would avoid "now" here and simply say "Open the following URL". Tutorials are comprised of many "causally-related" steps. Using "segway" phrases such as "now" or "then" or "after doing that" etc. is IMO unnecessary because doing so adds more words without improving clarity, and also tends to become repetitive very quickly.

## Alertmanager

Execute the command to port forward alertmanager locally on port `9093`.

```
kubectl -n monitoring port-forward svc/prometheus-operator-alertmanager 9093:9093
```

Now open the following URL: [http://localhost:9093](http://localhost:9093).

## Grafana

### Using Port Forward

Execute the command to port forward grafana dashboard locally on port `8080`.

```
kubectl -n monitoring port-forward svc/prometheus-operator-grafana 8080:80
```

> Obtain grafana `admin` user password by running the following command:

Reviewer: ```suggestion
Obtain the password for the `admin` Grafana user by running the following command:
```

- Grafana should be capitalized.
- Missing article.
- "Grafana admin user password" is IMO confusing as it's just a series of nouns and it's not clear what we are talking about. My suggestion above makes it clearer that we are talking about "the password which belongs to the Grafana user named `admin`".

```
kubectl -n monitoring get secret prometheus-operator-grafana -o jsonpath='{.data.admin-password}' | base64 -d && echo
```

> Now open the following URL: [http://localhost:8080](http://localhost:8080). Enter username `admin` and password obtained from the previous step.

Reviewer: Articles.

### Using Ingress

> To expose grafana to the internet using Ingress, provide the host field. The configuration for grafana in prometheus-operator should look like this:

Reviewer: What does "Ingress" refer to? Do we mean "using an Ingress k8s resource" / "using a ingress controller"? I feel we need to be a bit more elaborate here.

Author: using ingress kubernetes resource, hence it is capitalised.

Reviewer: OK. This wasn't clear to me from reading the above.

> To expose grafana to the internet using Ingress, provide the host field. The configuration for grafana in prometheus-operator should look like this:

Reviewer: ```suggestion
To expose grafana to the internet using Ingress, provide the host field. The configuration for grafana in prometheus-operator should look like the following:
```

"like this" is too informal IMO and doesn't fit the tone used in the rest of the guide.

# Access Monitoring on Lokomotive

## Contents

* [Introduction](#introduction)

> * [Presumptions](#presumptions)

Reviewer 2: ```suggestion
* [Prerequisites](#prerequisites)
```

* [Prometheus](#prometheus)
* [Alertmanager](#alertmanager)
* [Grafana](#grafana)

>   * [Using Port Forward](#using-port-forward)
>   * [Using Ingress](#using-ingress)

Reviewer 2: ```suggestion
* [Using Port Forward](#using-port-forward)
* [Using Ingress](#using-ingress)
```

Author: They are sub section, why remove the indentation?

Reviewer 2: my bad.

## Introduction

You can deploy the monitoring stack shipped with lokomotive as shown in the Prometheus Operator component [guide](../configuration-reference/components/prometheus-operator.md).

> ## Presumptions

Reviewer 2: ```suggestion
## Prerequisites
```

- Prometheus Operator component is deployed in `monitoring` namespace. If you have specified a different namespace in config then change the commands accordingly, `kubectl -n <your namespace> ...`.

## Prometheus

> Run following command to port forward Prometheus locally on port `9090`.

Reviewer 2: ```suggestion
Execute the command to port forward Prometheus locally on port `9090`:
```

s/Run following command/Execute the command/g

```
kubectl -n monitoring port-forward svc/prometheus-operator-prometheus 9090:9090
```

Now goto [http://localhost:9090](http://localhost:9090).

## Alertmanager

Run following command to port forward Alertmanager locally on port `9093`.

```
kubectl -n monitoring port-forward svc/prometheus-operator-alertmanager 9093:9093
```

> Now goto [http://localhost:9093](http://localhost:9093).

Reviewer 2: Is `Now goto` the correct words here ? Perhaps something like `Access the URL` ?

Reviewer 3: Maybe `Now open the following URL in your browser?`

## Grafana

### Using Port Forward

Run following command to port forward Grafana dashboard locally on port `8080`.

```
kubectl -n monitoring port-forward svc/prometheus-operator-grafana 8080:80
```

Obtain Grafana `admin` user password by running following command:

```
kubectl -n monitoring get secret prometheus-operator-grafana -o jsonpath='{.data.admin-password}' | base64 -d && echo
```

Now goto [http://localhost:8080](http://localhost:8080). Enter username `admin` and password obtained from previous step.

### Using Ingress

> Make sure you are using a configuration for Grafana in Prometheus Operator something similar to following:

Reviewer 2: I am not sure about this block. 
What else could be the difference in the configuration ? Its a configuration of our component.

Am I right in saying that you want the the user to use the correct `grafana.ingress.host` value in the component and also when following the docs ? Why not point to the `prometheus-operator` component configuration reference documentation ?

Author: Since this is how to guide, we are talking about scenarios. By that logic we can always point folks to the config ref doc to do anything.

The motivation of this doc is to show what various combination of config can allow user to do.

```tf
component "prometheus-operator" {
  grafana {
    ingress {
      host = "grafana.mydomain.com"
    }
  }
}
```

> **NOTE**: If you are running this component on Packet then make sure that you have made a DNS entry for `grafana.mydomain.com` against the Packet EIP.

Reviewer: I feel we should ensure Packet isn't a "special case" rather than mention it as a special case here. Maybe it's time to address https://github.com/kinvolk/lokomotive/issues/470?

Author: I can put the information about AWS as well. Like it is automatically available, just add the ingress host as `grafana.<cluster name>.<domain>`.

Reviewer: Again, IMO Packet should have a similar UX to AWS. But I understand that treating the feature parity problems is out of scope for this PR.

Author: This is modified, PTAL.
